TeaDiary/Scripts/App/validation.js

```diff
@@ -0,0 +1,12 @@
+﻿function isTeaValid(tea) {
+    if (!tea) {
+        return false;
+    }
+    if (!tea.Name || !tea.Type) {
+        return false;
+    }
+    if (tea.Name === "" || tea.Type === "") {
```

Comment:
I would rather group them by fields. Like `if (name == null ! name === "") {} if (type == null || type === "") ...`

Reply:
`x == null` is the same as `x === null || x === undefined`. It's the only way when using `==` is allowed without shooting yourself in the foot.